libraries/abstractions/common_io/test/test_iot_uart.c

```diff
@@ -0,0 +1,817 @@
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTIoctlFuzzing );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTReadSyncFuzzing );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTReadAsyncFuzzing );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTWriteSyncFuzzing );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTWriteAsyncFuzzing );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTCancel );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTIoctlWhenBusy )
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTWriteAsyncReadAsyncLoopbackTest );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTWriteSyncReadAsyncLoopbackTest );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTIoctlGetSet );
+    RUN_TEST_CASE( TEST_IOT_UART, AFQP_IotUARTOpenCloseCancelFuzzing );
+}
+/*-----------------------------------------------------------*/
+/*-----------------------------------------------------------*/
+
+/**
+ * hardware loopback. The Tx and Rx pin on the vendor board are shorted with a
+ * connector and the test is run by transmitting some bytes on write sync and checking
+ * if same characters are read synchronously.
+ */
+TEST( TEST_IOT_UART, AFQP_AssistedIotUARTWriteReadSync )
+{
+    uint8_t cpBuffer[ testIotUARTBUFFERSIZE ] = { 0 };
+    uint8_t cpBufferRead[ testIotUARTBUFFERSIZE ] = { 0 };
+
+    strncpy( ( char * ) cpBuffer, ( char * ) testIotUART_BUFFER_STRING, testIotUART_BUFFER_LENGTH );
+    IotUARTHandle_t xUartHandle;
+    int32_t lRead, lWrite, lClose;
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+        lWrite = iot_uart_write_sync( xUartHandle, cpBuffer, testIotUART_BUFFER_LENGTH );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lWrite );
+
+        lRead = iot_uart_read_sync( xUartHandle, cpBufferRead, testIotUART_BUFFER_LENGTH );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lRead );
+
+        TEST_ASSERT_EQUAL( 0, strncmp( ( char * ) cpBuffer, ( char * ) cpBufferRead, testIotUART_BUFFER_LENGTH ) );
+    }
+
+    lClose = iot_uart_close( xUartHandle );
+    TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lClose );
+}
+/*-----------------------------------------------------------*/
+/*-----------------------------------------------------------*/
+/**
+ * @brief Test Function to check UART Ioctl API specifications.
+ */
+TEST( TEST_IOT_UART, AFQP_IotUARTIoctlGetSet )
+{
+    IotUARTHandle_t xUartHandle;
+    int32_t lIoctl, lClose, lTransferAmount;
+    IotUARTConfig_t pvgetConfigBuffer, pvConfigBuffer, pvOriginalConfig;
+
+    xUartHandle = iot_uart_open( ustestIotUartPort );
+    TEST_ASSERT_NOT_EQUAL( NULL, xUartHandle );
+
+    if( TEST_PROTECT() )
+    {
+
+        /* Save original UART config */
+        lIoctl = iot_uart_ioctl( xUartHandle, eUartGetConfig, &pvOriginalConfig );
+        TEST_ASSERT_EQUAL( IOT_UART_SUCCESS, lIoctl );
+        TEST_ASSERT_NOT_EQUAL( testIotUART_TEST_BAUD_RATE, pvOriginalConfig.ulBaudrate );
```

Comment:
Why the original config is checked here? How can you ensure it is with the default values?

Reply:
This is a rebase issue. We should not check any files in but the assisted test scripts. I will make a task of rebasing and reopening the PR.